### PR TITLE
fix cloning of DB connection for sqlite in-memory db

### LINKED ISSUE #14131: BC cloning DB does not keep DB's state with SQLite
As discussed with @cebe, this topic is opened to address an issue when cloing a Sqlite DB and expecting it to have the original state. Somehow this is broken and I drilled it down to commit: 
https://github.com/cebe/yii2/commit/5ed6910e1cca13c3fea677a564dc13a36533f50c

and in specific because of:
```php
$this->pdo = null;
```

Hiding this works and enables me to clone & reuse data of a SqlLite database for testing purposes

### PR DESCRIPTION
**Pull request by cebe**
fixes #14131

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14131

### CODE REVIEW

**Comment by cebe** [comment]
@dynasource please check if this fixes your issue.

**Comment by dynasource** [comment]
just checked. This patch seems to fix the issue. Thanks!

**Review by dynasource — APPROVED** [review]

File: framework/db/Connection.php [hunk 1]
```diff
@@ -1090,8 +1090,11 @@ public function __clone()
 
         $this->_master = false;
         $this->_slave = false;
-        $this->pdo = null;
         $this->_schema = null;
         $this->_transaction = null;
+        if (strpos($this->dsn, 'sqlite::memory:') !== 0) {
```
Comment (samdark) [inline framework/db/Connection.php]: I'd use http://php.net/manual/en/function.strncmp.php instead.

**Review by samdark — APPROVED** [review]

File: framework/db/Connection.php (hunk 1 above)
Comment (cebe) [inline framework/db/Connection.php]: right, seems to be faster: https://3v4l.org/MckkL

**Pull request closed without merging**